src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -140,7 +141,21 @@ protected void Initialize(IDiscoveryContext context, IMessageLogger messageLogge
             }
         }
 
-        
+        static readonly List<string> ForbiddenFolders = new List<string> { "windows", "program files (x86)", "program files", "programdata" };
```

Comment:
I think I'd prefer to do a case insensitive StartsWith comparison between `Directory.GetCurrentDirectory()` (always absolute) and `Path.GetFullPath(Environment.GetFolderPath(...))` for each of these things.

@vpenades also asked about `C:\Users` with the exception of `Path.GetTempPath()`; what do you think of that?

Reply:
Instead of using a plain list of paths, like "windows", "program files (x86)", shouldn't be better to use `System.Environment.GetSpecialFolderPath`?

Also, although it's rarely done, users can change windows and program files location, also, it's sometimes localized, for example, "program files (x86)" is "archivos de programa (x86)" in spanish.

So I think it's better to ask the OS for these folders.

Reply:
@vpenades  Good point!  
@jnm2 There are so many places below Users where it would be legal to use, so I could not really see what to exclude there that made sense.  Did I overlook something?

Reply:
> There are so many places below Users where it would be legal to use, so I could not really see what to exclude there that made sense.

This is a good point. Let's not guess.

%programdata% seems just as acceptable to write into as %appdata% or %localappdata%.

Reversing direction, maybe for now only block it if it's under `Environment.GetFolderPath(Environment.SpecialFolder.Windows)`?

Reply:
@jnm2 I agree it's fine to allow users and programs data.

I would disallow everything under windows and programs files x86 and x64

Reply:
Under .net 3.5 which we are using, we don't have x86 program files, and we don't have windows.   We can look using environment variables, but then again, I would expect those to be locale sensitives.

Reply:
@OsirisTerje 

I am not sure about .net3.5 ... the problem is not only about localization; users can choose to install windows or program files directory in a path different than the default (or even change the drive unit)

Getting the directories on net core... the only solution I could find around is by using [Microsoft.Windows.Compatibility package](https://www.nuget.org/packages/Microsoft.Windows.Compatibility) which I guess it will work under windows, and throw exceptions or return null under other OS.

Reply:
@OsirisTerje This is a separate thing, but the minimum version of .NET Core that Microsoft supports as of a month ago is .NET Core 2.1. It would make life easier if we no longer supported anything earlier as well.

Reply:
I would be very surprised if environment variable *names* were localized; if they are, what's the point of having them? I would expect environment variable *values* to be localized.

Reply:
@jnm2  We should have some survey out for stuff like this.  Or some metrics.......